### PR TITLE
Settings view added

### PR DESCRIPTION
**Pull request by andre2007**
This pull request adds a new site "Settings" which describes the values you can set in settings.json. E.g. defaultCompiler, defaultArchitecture...

### CODE REVIEW

**Comment by andre2007** [comment]
@wilzbach I am not 100% sure about the etc paths. In change log https://github.com/dlang/dub/pull/1459/files you mentioned it as fixed path (/etc/...) while in coding it is related to the dub executable.
Also etc on Windows seems somehow strange.

In the pull request the etc paths are described according to the implementation.

**Comment by wilzbach** [comment]
The interesting bit is here:

https://github.com/dlang/dub/blob/ed38ec747ab0088e77e900af6dcca4058a10da6b/source/dub/dub.d#L293

```d
m_config = new DubConfig(jsonFromFile(NativePath(thisExePath).parentPath ~ "../etc/dub/settings.json", true), m_config)
```

This means that when dub is installed as `/usr/bin/dub` (i.e. globally) it will use `/etc/dub/settings.json`

File: views/settings.dt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		|{
+		|	"defaultArchitecture": "x86_64",
+		|	"defaultCompiler": "ldc"
+		|}
+
+	p The file settings.json can be located in different locations. Last item in list has highest priority.
+	
+	p Windows
+		ul
+			li <code>%ProgramData%\dub\setting.json</code>
```
Comment (wilzbach) [inline views/settings.dt]: settings.json

File: views/settings.dt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+			li <code>&lt;dub executable folder&gt;\..\etc\dub\settings.json</code>
+			li <code>%APPDATA%\dub\settings.json</code>
+
+	p Posix
+		ul
+			li <code>/var/lib/dub/setting.json</code>
```
Comment (wilzbach) [inline views/settings.dt]: settings.json

File: views/settings.dt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+			li <code>&lt;dub executable folder&gt;/../etc/dub/settings.json</code>
+			li <code>~/.dub/setting.json</code>
```
Comment (wilzbach) [inline views/settings.dt]: settings.json

File: views/settings.dt (hunk 1 above)
Comment (andre2007) [reply views/settings.dt]: Fixed

**Comment by andre2007** [comment]
Ping @wilzbach 
As also written in https://github.com/dlang/dub/issues/1472 the description in this pull is according the implementation but maybe the implementation  is wrong:

```
My assumption is, current logic will find for /usr/bin/dub the settings file /usr/etc/dub/settings.json and not /etc/dub/settings.json.
I validated my assumption on a linux system which has /usr/bin/dub by creating file /usr/etc/dub/settings.json with invalid JSON format. This file is always found by dub.

NativePath(thisExePath).parentPath will find path /usr/bin
../etc/dub/settings.json will go 1 hierarchy up and the into folder /etc/dub
```

How to go forward with the new settings page?

**Comment by andre2007** [comment]
@wilzbach  @MartinNowak How to continue with this pull request?

**Comment by thewilsonator** [comment]
What needs doing?

**Comment by andre2007** [comment]
The settings are described as they are implemented. 

There is some confusion as the etc path is not implemented  as someone would expect. But I cannot judge on this as I have not much Linux experience. 

I suggest to merge this pr and if the etc path implementation is changed, this change has to be reflected in this documentation

**Comment by andre2007** [comment]
Ping :)

**Pull request merged**